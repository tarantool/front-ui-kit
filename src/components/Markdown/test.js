import React from 'react';
import renderer from 'react-test-renderer';
import { Markdown } from './index';

describe('Markdown', () => {
  it('renders code inline', () => {
    const tree = renderer.create(<Markdown text='`Code line`' />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders code block', () => {
    const md =
`\`\`\`
Code block
\`\`\``;

    const tree = renderer.create(<Markdown text={md} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders mixed content', () => {
    const instruction = `
## Connect to Tarantool Cartridge using [python client](https://github.com/tarantool/tarantool-python)

First, **install** *tarantool* package using *pip3*:

\`\`\`bash
pip3 install tarantool
\`\`\`

Create a file example.py with the code to get started:

\`\`\`python
from tarantool import Connection
c = Connection(
    "try-cartridge.tarantool.io", 
    10300, 
    user='admin', 
    password='npngatmwsf'
)
result = c.insert("examples", (333, 'Value', 'Value'))
space = c.space("examples")
results = space.select(333)
print(results)
\`\`\`

Run the script

\`\`\`bash
python3 example.py
\`\`\`
`;

    const tree = renderer.create(<Markdown text={instruction} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders with built-in component overriding', () => {
    const md =
`
# Bee careful

Sed ut perspiciatis, unde omnis iste natus error sit voluptatem accusantium doloremque laudantium

* Create file
* Rename file
* Save file
`;

    const tree = renderer.create(
      <Markdown
        overrides={{
          ul: ({ children, ...props }) => (
            <ul {...props} className='iAmOverridedComponent'>{children}</ul>
          )
        }}
        text={md}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

});
