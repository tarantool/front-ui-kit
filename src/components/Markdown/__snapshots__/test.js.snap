// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Markdown renders code block 1`] = `
<div
  className="css-1fcvszc"
>
  <pre>
    <code
      className="css-19aciib"
    >
      Code block
    </code>
  </pre>
</div>
`;

exports[`Markdown renders code inline 1`] = `
<div
  className="css-1fcvszc"
>
  <p
    className="css-1eitoz7"
  >
    <code
      className="css-19aciib"
    >
      Code line
    </code>
  </p>
</div>
`;

exports[`Markdown renders mixed content 1`] = `
<div
  className="css-1fcvszc"
>
  <div>
    <h2
      className="css-v70fzj"
      id="connect-to-tarantool-cartridge-using-python-clienthttpsgithubcomtarantooltarantool-python"
    >
      Connect to Tarantool Cartridge using 
      <a
        className="css-kzn11m"
        href="https://github.com/tarantool/tarantool-python"
        target="_blank"
      >
        python client
      </a>
    </h2>
    <p
      className="css-1eitoz7"
    >
      First, 
      <strong>
        install
      </strong>
       
      <em>
        tarantool
      </em>
       package using 
      <em>
        pip3
      </em>
      :
    </p>
    <pre>
      <code
        className="language-bash css-3fhyop"
      >
        pip3 install tarantool
      </code>
    </pre>
    <p
      className="css-1eitoz7"
    >
      Create a file example.py with the code to get started:
    </p>
    <pre>
      <code
        className="language-python css-3fhyop"
      >
        from tarantool import Connection
c = Connection(
    "try-cartridge.tarantool.io", 
    10300, 
    user='admin', 
    password='npngatmwsf'
)
result = c.insert("examples", (333, 'Value', 'Value'))
space = c.space("examples")
results = space.select(333)
print(results)
      </code>
    </pre>
    <p
      className="css-1eitoz7"
    >
      Run the script
    </p>
    <pre>
      <code
        className="language-bash css-3fhyop"
      >
        python3 example.py
      </code>
    </pre>
  </div>
</div>
`;
